public/app/features/dashboard/services/DashboardSrv.ts

```diff
@@ -5,6 +5,24 @@ import { DashboardMeta } from 'app/types';
 import { getBackendSrv } from 'app/core/services/backend_srv';
 import { saveDashboard } from 'app/features/manage-dashboards/state/actions';
 import { RemovePanelEvent } from '../../../types/events';
+import { BackendSrvRequest } from '@grafana/runtime';
+
+export interface SaveDashboardOptions {
+  /** The complete dashboard model. If `dashboard.id` is not set a new dashboard will be created. */
+  dashboard: DashboardModel;
```

Comment:
If a `DashboardModel` is accepted as a param, I think `dashboard.getSaveModelClone` might have to be called and its return value used

Reply:
good point, change it in cf3ddc9068